go.mod

```diff
@@ -1,25 +1,25 @@
 module github.com/strangelove-ventures/ibctest
 
-go 1.17
+go 1.18
```

Comment:
Because ibctest is used as a library, we generally support the N and N-1 release. Therefore, we purposefully keep go at 1.17 so projects on 1.17 can use ibctest as a library.

However, the cosmos SDK and a sample of chains appear to be on 1.18. We may need to verify majority of chains are on 1.18. 

So it may be safe to make this upgrade although it violates our current version policy. 

I'll do some more verification on the current state of chain codebases.

Reply:
I created this [quick and dirty script](https://gist.github.com/DavidNix/26c9e86b7941225ec6576d2a858983bf) to get a sense of the Go versions for all the chains. 

The results are:

```
go 1.17 terra-money/core
go 1.16 Decentr-net/decentr
go 1.17 crypto-org-chain/chain-main
go unknown CudoVentures/cudos-builders
go unknown Sifchain/sifnode.git
go 1.18 IDEP-network/ion
go unknown /
go 1.17 memecosmos/meme
go 1.16 ChronicNetwork/cht
go 1.17 microtick/mtzone
go 1.14 Agoric/ag0
go 1.18 umee-network/umee
go unknown Switcheo/carbon-bootstrap
go 1.18 public-awesome/stargaze
go 1.17 vincadian/arkh-blockchain
go 1.16 bandprotocol/chain
go 1.17 alpha-omega-labs/genesisd
go 1.18 comdex-official/comdex
go 1.13 thorchain/thornode
go 1.17 ChihuahuaChain/chihuahua
go 1.17 ovrclk/akash
go 1.18 Team-Kujira/core
go unknown imversed/imversed
go 1.17 stateset/core
go unknown saage-tech/saage-testnet-1
go 1.18 cosmos/gaia
go 1.18 cosmos/gaia
go 1.17 kava-Labs/kava
go 1.16 iov-one/starnamed
go 1.18 axelarnetwork/axelar-core
go unknown /
go 1.15 PeggyJV/sommelier
go 1.17 ixofoundation/ixo-blockchain
go 1.16 galaxies-labs/galaxy
go 1.17 echelonfoundation/echelon
go 1.16 KiFoundation/ki-tools
go 1.17 cybercongress/go-cyber
go unknown nomic-io/nomic
go 1.17 bitsongofficial/go-bitsong
go 1.17 fetchai/fetchd
go unknown AssetMantle/node.git
go 1.17 sentinel-official/hub
go 1.18 osmosis-labs/osmosis
go 1.17 BitCannaGlobal/bcna
go 1.17 rizon-world/rizon
go 1.17 confio/tgrade
go unknown /
go 1.17 provenance-io/provenance
go 1.18 evmos/evmos
go 1.18 notional-labs/dig
go unknown /
go 1.17 medibloc/panacea-core
go 1.18 persistenceOne/persistenceCore
go unknown Gravity-Bridge/Gravity-Bridge
go 1.18 Team-Kujira/core
go 1.17 crescent-network/crescent
go 1.18 likecoin/likecoin-chain
go 1.17 scrtlabs/SecretNetwork
go 1.16 cerberus-zone/cerberus
go 1.18 lum-network/chain
go unknown InjectiveLabs/injective-chain-releases
go 1.16 vidulum/mainnet
go 1.15 e-money/em-ledger
go 1.15 oraichain/orai
go 1.17 cheqd/cheqd-node
go unknown ODIN-PROTOCOL/networks
go 1.17 irisnet/irishub
go 1.17 crypto-org-chain/cronos
go 1.18 CosmosContracts/juno
go 1.17 desmos-labs/desmos
go 1.17 konstellation/konstellation
go 1.17 terra-money/core
go 1.18 regen-network/regen-ledger
go 1.17 firmachain/firmachain
go 1.18 ShentuChain/shentu

Groups
unknown 14
1.13 1
1.14 1
1.15 3
1.16 8
1.17 30
1.18 18
```

It seems a fair amount (30 chains) are still using 1.17 including (what I think) are more popular ones like Kava, Akash, and Sentinel.

Reply:
Go 1.19 [has an RC out](https://golangtutorial.dev/news/g-1.19rc1-released/) expected to be released in August.

I could be persuaded to upgrade to 1.18. The SDK upgraded to 1.18 as of 3 months ago. The first release requiring 1.18 was 3 months ago. 

```
❯ git tag --contains 9bed6333fab93dd0cda88d8c8c4b94beaf74146e
api/v0.1.0-alpha8
container/v1.0.0-alpha.3
container/v1.0.0-alpha.4
core/v0.1.0
depinject/v1.0.0-alpha.1
errors/v1.0.0-beta.5
errors/v1.0.0-beta.6
errors/v1.0.0-beta.7
math/v1.0.0-beta.1
math/v1.0.0-beta.2
orm/v1.0.0-alpha.12
v0.46.0-rc1
v0.46.0-rc2
```

If I'm reading this correctly, it appears v0.46 is the first release with 1.18. 

I'm using the Cosmos SDK as a loose guide to determine when chains should upgrade.